spread-shellcheck

```diff
@@ -133,15 +137,59 @@ def checksection(data, env: Dict[str, str]):
                             stdout=subprocess.PIPE,
                             stdin=subprocess.PIPE,
                             shell=True)
-    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=30)
+    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=60)
     if proc.returncode != 0:
         raise ShellcheckRunError(stdout)
 
 
+class Cacher:
+    _instance = None
+    def __new__(cls):
+        if cls._instance is not None:
+            return cls._instance
+        instance = super().__new__(cls)
+        cls._instance = instance
+        instance.enabled = True
+        return instance
+
+    def disable(self):
+        logging.debug("caching is disabled")
+        self.enabled = False
+
+    @staticmethod
+    def _cache_path_for(digest):
+        prefix = digest[:2]
+        return Path.home().joinpath(".cache", "spread-shellcheck", prefix, digest)
+
+    def is_cached(self, data, path):
+        h = hashlib.sha256()
+        h.update(data)
+        hdg = binascii.b2a_hex(h.digest()).decode()
+        if not self.enabled:
+            return False, hdg
```

Comment:
Why not do this check at the beginning of the method and return an empty string instead of hgd? Seems simpler and would avoid computing the digest unnecessarily